### PR TITLE
Adjust code editor active line background and cursor color.

### LINKED ISSUE #18717: Pipeline rule editor cursor is invisible on the light theme
In Graylog 6.0.0-beta.2+8b77fdb, the pipeline rule source code editor text box cursor is nearly invisible on the light theme. It's visible on the dark theme. This may be happening in other text boxes too. 

https://github.com/Graylog2/graylog2-server/assets/2953243/58ae065a-fa5e-41bd-b950-c4010362fbe7

## Your Environment

- Graylog Version: Graylog 6.0.0-beta.2+8b77fdb
- Java Version: Embedded
- OpenSearch Version: 2.12.0
- MongoDB Version: 7.0.6
- Operating System: Debian 12
- Browser version: Version 122.0.6261.129 (Official Build) (arm64)

### LINKED ISSUE #18847: Color contrast problems in Ace source code editor
## Description

The contrast for highlighted lines in the Ace editor (e.g. pipeline rule editor) is problematic in the light theme.

In this example, the `set_field` text is really hard to see. (at least for me :smile:)

image

## Steps to Reproduce (for bugs)

1. Switch to the light theme
2. Open the pipeline editor for a rule
3. Look at different highlighted lines in a rule

## Your Environment

* Graylog Version: 6.0.0-beta.3

### PR DESCRIPTION
**Pull request by ousmaneo**
Before these changes, we had issues with the active line background color. We also adjust the cursor color for other usage.

fix #18847 
fix #18717

/nocl

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-web-interface/src/components/common/SourceCodeEditor.tsx [hunk 1]
```diff
@@ -45,6 +45,22 @@ const SourceCodeContainer = styled.div<ContainerProps>(({ $resizable, theme }) =
...
+    .ace_gutter-active-line {
+      background-color: ${theme.colors.variant.lightest.default};
+    }
+    .ace_gutter {
+      background: ${theme.colors.variant.lighter.default};
+    }
+    .ace_cursor {
+      border-color: ${theme.colors.global.textDefault};
+    }
+  }
```
Comment (linuspahl) [inline graylog2-web-interface/src/components/common/SourceCodeEditor.tsx:63]: @ousmaneo @grotlue I am having a look at https://github.com/Graylog2/graylog2-server/issues/19346 & https://github.com/Graylog2/graylog2-server/issues/19695, which is about the color of comments in the pipeline editor.

To fix it we just just need to define a suitable color for `.ace_comment`. The current color is defined in https://github.com/Graylog2/sawmill/blob/main/lib/styled-components/component-styles/aceEditor.ts.

The `aceEditor.ts` is implemented here via `theme.components.aceEditor`.

Years ago we moved these styles into the theme. https://github.com/Graylog2/graylog2-server/pull/8576 If I remember correctly we did this, so these styles are not regenerated on every keypress.

I lean towards changing the color for `.ace_cursor` in the `aceEditor.ts`, but I was wondering if there was a reason to not add the styling above to the `aceEditor.ts` in sawmill as well? The file already contains styling for all the classes we define above.

Comment (linuspahl) [reply graylog2-web-interface/src/components/common/SourceCodeEditor.tsx:63]: Or in other words do you see any disadvantage in moving these styles into the theme as well?

Comment (ousmaneo) [reply graylog2-web-interface/src/components/common/SourceCodeEditor.tsx:63]: Hi @linuspahl, I think it's fine if we move it to the theme.

Comment (linuspahl) [reply graylog2-web-interface/src/components/common/SourceCodeEditor.tsx:63]: Ok, great. Thanks.

**Pull request merged**